d2compiler/compile.go

```diff
@@ -998,6 +999,22 @@ func (c *compiler) validateKey(obj *d2graph.Object, f *d2ir.Field) {
 	}
 }
 
+func (c *compiler) validateLabels(g *d2graph.Graph) {
+	for _, obj := range g.Objects {
+		if obj.Shape.Value != d2target.ShapeText {
+			continue
+		}
+		if obj.Attributes.Language != "" {
+			// blockstrings have already been validated
+			continue
+		}
+		if strings.TrimSpace(obj.Label.Value) == "" {
+			c.errorf(obj.Label.MapKey, "shape text must have a non-empty label")
```

Comment:
wait why? https://play.d2lang.com/?script=qrBSUFLiAgQAAP__&

Reply:
This is just for `shape: text` which only makes sense with a label

Reply:
ohhh